### PR TITLE
ADAPT-2541: Added oclif framework, and command for cloning storyblok spaces. 

### PR DESCRIPTION
**Pull request by nplowman**
## Summary

- Installs OCLIF framework in the repo. (This will be used as the foundation for all CLI commands added to this repo going forward).
- Adds command for cloning content and structure of an existing Storyblok space to a new space.

## Review Tasks
To run this locally,
1. Checkout out this branch.
2. Run npm install
3. Note that when testing locally, you'll use ./node/run as the base command. (This will become `adapt-utils`once it is published on npm, and you install using `npm install -g adapt-dev-utils`)
4. Below are a few different ways you can run the command:
```
# Clone using name of source repo.
./node/run sb-clone --sourceName="OOD Giving" --targetName="cloned space test" --token=[your_token]
# Clone using numeric ID of source repo.
./node/run sb-clone --sourceId=78141 --targetName="cloned space test" --token=[your_token]
# Pass token in using environment variable instead of inline.
STORYBLOK_TOKEN=[your_token]
./node/run sb-clone --sourceName="OOD Giving" --targetName="cloned space test"
```
5. After you've cloned the space, you'll want to log into Storyblok and delete the test space.

## Notes
- You can get a token by logging into Storyblok and looking at the "My Account" page. 
- Documentation on the OCLIF framework can be found here: https://oclif.io/docs/introduction
- For a proper installation, we'll need to publish this to NPM. 
- Since this will be the first pull request using the new framework, it would also be good explore the documentation and help features of the CLI tool itself. For example:
```
# Run the base command, to see the command listing and top-level help.
./node/run
# See command-level help.
./node/run/sb-clone --help
```

### CODE REVIEW

**Comment by sherakama** [comment]
I've had a review of this tool and have been successful in cloning a space. I like the oclif library and I think that will be a good choice for us going forward. I had to add the `bin` dir in order to get it to work for me. Did I miss a step somewhere?

**Comment by nplowman** [comment]
@sherakama Let me look into that.... 
Might be an issue with the stock .gitignore folder that was in the repo excluding it bin from the repo.
Or there might be an additional build command that needs to be ran locally after cloning.

**Comment by nplowman** [comment]
@sherakama Looks like it was a .gitignore issue. I switch to the stock .gitignore that ships with oclif and added the missing files.

File: src/commands/config-set.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    {name: 'key'},
+    {name: 'value'}
+  ]
+
+  async run() {
+    const {args} = this.parse(ConfigSetCommand)
+    const configFilePath = path.join(this.config.configDir, 'config.json')
+
+    // Initialize config file if it does not yet exit.
+    if (!fs.existsSync(configFilePath)) {
```
Comment (sherakama) [inline src/commands/config-set.js:16]: This gives me an error message in the console when the file does not exist but it succeeds in creating it and filling the values. 

```(node:45493) [ENOENT] Error Plugin: adapt-dev-utils: /Users/admin/.config/adapt-dev-utils/config.json: ENOENT: no such file or directory, open '/Users/admin/.config/adapt-dev-utils/config.json'```

**Review by sherakama — APPROVED** [review]
This is working great. Going to merge this down and we can tackle bugs revisions in future PRs/Tickets.

**Comment by sherakama** [comment]
Ooo, the prompts are very nice.

**Pull request merged**